prefect_dbt/cli/configs/base.py

```diff
@@ -0,0 +1,105 @@
+"""Module containing models for base configs"""
+
+from typing import Any, Dict, Optional
+
+from prefect.blocks.core import Block
+from pydantic import Field
+
+
+class DbtConfigs(Block):
+    """
+    Abstract class for other dbt Configs.
+
+    Args:
+        extras: Extra target configs' keywords, not yet added
+            to prefect-dbt, but available in dbt.
+    """
+
+    _is_anonymous = True
+
+    extras: Optional[Dict[str, Any]] = None
+
+    def get_configs(self):
+        """
+        Helper function to parse configs.
+        """
+        configs_json = {}
+        for key, val in self.dict().items():
+            if val is not None:
+                if key in ["extras", "credentials"]:
+                    configs_json.update(**val)
+                else:
+                    configs_json[key.rstrip("_")] = val
+        return configs_json
+
+
+class TargetConfigs(DbtConfigs):
+    """
+    Target configs contain credentials and
+    settings, specific to the warehouse you're connecting to.
+    This is the base dbt TargetConfigs model;
+    ideally use the TargetConfigs containing authentication keywords, e.g.
+    `prefect_dbt.cli.snowflake.SnowflakeUserPasswordTargetConfigs`,
+    but if the desired TargetConfigs is missing, use this class with
+    extra keywords, or help contribute a pull request.
+    To find valid keys, head to the [Available adapters](
+    https://docs.getdbt.com/docs/available-adapters) page and
+    click the desired adapter's "Profile Setup" hyperlink.
+
+    Args:
+        type: The name of the database warehouse
+        schema: The schema that dbt will build objects into;
+            in BigQuery, a schema is actually a dataset.
+        threads: The number of threads representing the max number
+            of paths through the graph dbt may work on at once.
+    """
+
+    type: str
+    schema_: str = Field(alias="schema")
```

Comment:
@desertaxle after I made this inherit from Block, my tests are failing due to a weird error; I suspect Block class does something funky with pydantic.Field objects and drops them?

Minimal example:
```
from prefect_dbt.cli.configs import GlobalConfigs, TargetConfigs
from prefect_dbt.cli import DbtCliProfile
from prefect_dbt.cli.commands import trigger_dbt_cli_command
from prefect import flow

target_configs = TargetConfigs(type="abc", schema="abcdef")

dbt_cli_profile_bare = DbtCliProfile(
    name="prefecto",
    target="testing",
    target_configs=target_configs,
)


@flow
def test_flow():
    return trigger_dbt_cli_command(
        "dbt ls", profiles_dir=".", dbt_cli_profile=dbt_cli_profile_bare
    )
test_flow()
```

Outputs
```
ValidationError: 1 validation error for TargetConfigs
schema
  field required (type=value_error.missing)
```

I tried specifying `schema_` and `schema` to no avail.

Reply:
```

  File "/Users/andrew/Applications/python/orion/src/prefect/task_runners.py", line 207, in submit
    result = await run_fn(**run_kwargs)
  File "/Users/andrew/Applications/python/orion/src/prefect/engine.py", line 873, in begin_task_run
    return await orchestrate_task_run(
  File "/Users/andrew/Applications/python/orion/src/prefect/engine.py", line 934, in orchestrate_task_run
    resolved_parameters = await resolve_inputs(parameters)
  File "/Users/andrew/Applications/python/orion/src/prefect/engine.py", line 1150, in resolve_inputs
    return await visit_collection(
  File "/Users/andrew/Applications/python/orion/src/prefect/utilities/collections.py", line 286, in visit_collection
    values = await gather(*[visit_nested(v) for v in values])
  File "/Users/andrew/Applications/python/orion/src/prefect/utilities/asyncutils.py", line 341, in gather
    keys.append(tg.start_soon(call))
  File "/Users/andrew/Applications/python/orion/src/prefect/utilities/asyncutils.py", line 317, in __aexit__
    retval = await self._task_group.__aexit__(*tb)
  File "/Users/andrew/mambaforge/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 574, in __aexit__
    raise exceptions[0]
  File "/Users/andrew/Applications/python/orion/src/prefect/utilities/asyncutils.py", line 286, in _run_and_store
    self._results[key] = await fn(*args)
  File "/Users/andrew/Applications/python/orion/src/prefect/utilities/collections.py", line 301, in visit_collection
    items = await gather(
  File "/Users/andrew/Applications/python/orion/src/prefect/utilities/asyncutils.py", line 341, in gather
    keys.append(tg.start_soon(call))
  File "/Users/andrew/Applications/python/orion/src/prefect/utilities/asyncutils.py", line 317, in __aexit__
    retval = await self._task_group.__aexit__(*tb)
  File "/Users/andrew/mambaforge/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 574, in __aexit__
    raise exceptions[0]
  File "/Users/andrew/Applications/python/orion/src/prefect/utilities/asyncutils.py", line 286, in _run_and_store
    self._results[key] = await fn(*args)
  File "/Users/andrew/Applications/python/orion/src/prefect/utilities/collections.py", line 306, in visit_collection
    model_instance = typ(
  File "/Users/andrew/Applications/python/orion/src/prefect/blocks/core.py", line 94, in __init__
    super().__init__(*args, **kwargs)
  File "pydantic/main.py", line 331, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for TargetConfigs
```

Reply:
Workaround is just use `schema_` instead of `schema_ = Field(alias="schema")`